core/src/main/java/org/infinispan/security/impl/SubjectACL.java

```diff
@@ -0,0 +1,17 @@
+package org.infinispan.security.impl;
+
+import java.util.Set;
+
+/**
+ * SubjectRoles.
+ *
+ * @author Tristan Tarrant
+ * @since 8.1
+ */
+public class SubjectACL {
```

Comment:
Probably the best way here is to change the AclCache key to be the combination of Cache/Subject. At this point we can cache the mask